### DISCUSSION TITLE
Episode 155: Don't Hurt Yourself (with C++) with Jonathan O'Connor

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#47] [Episode]
Episode Link: https://adspthepodcast.com/2023/11/10/Episode-155.html

Feel free to write comments or ask questions 🙂

**Comment by jdebache** [comment]
C# is also a great multi-paradigm language these days. With every release, they both make it easier to write functional code (see all the extremely nifty [pattern matching](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching) for example) **and** expose lower level constructs for more control (see the [span type](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching)). Would be cool if you guys could have Mads Torgersen or Anders Hejlsberg on at some point. They've been on other podcasts of course, but the ADSP take is always unique :)
